### PR TITLE
[WIP] Use the new rustc interface

### PR DESCRIPTION
**Pull request by Zoxc**
Shows the changes required to compile with https://github.com/rust-lang/rust/pull/56732

### CODE REVIEW

File: src/bin/miri.rs [hunk 1]
```diff
@@ -7,119 +7,52 @@ extern crate rustc_metadata;
...
-    let (entry_def_id, _) = tcx.entry_fn(LOCAL_CRATE).expect("no main function found!");
+            miri::eval_main(tcx, entry_def_id, self.validate);
+        });
 
-    miri::eval_main(tcx, entry_def_id, validate);
+        compiler.session().abort_if_errors();
 
-    state.session.abort_if_errors();
+        // Continue execution
+        true
```
Comment (bjorn3) [inline src/bin/miri.rs]: > control.after_analysis.stop = Compilation::Stop;

This didn't continue execution before.

**Comment by RalfJung** [comment]
Thanks!

I think `src/bin/miri-rustc-tests.rs` probably needs similar updates?  You can build with `--all-features --all-targets` to make sure you build everything.

**Comment by Zoxc** [comment]
I've updated this so `--all-features --all-targets` builds.

**Comment by RalfJung** [comment]
@Zoxc this is still marked as WIP, is there anything left to do?

Also, there are conflicts. Let me know if you can take care of them or if I should.

**Comment by oli-obk** [comment]
rebased in https://github.com/rust-lang/miri/pull/659

**Comment by RalfJung** [comment]
Thanks! Closing in favor of #659 then.

**Pull request closed without merging**